pyteal/ast/maybe.py

```diff
@@ -32,12 +33,13 @@ def __init__(
             args (optional): Stack arguments for the op. Defaults to None.
         """
 
-        def local_version_check(option: "CompileOptions"):
+        def local_version_check(options: "CompileOptions"):
             verifyProgramVersion(
                 minVersion=op.min_version,
-                version=option.version,
+                version=options.version,
                 msg=f"{op.value} unavailable",
             )
+            compile_check(options)
```

Comment:
@jdtzmn I think you raise a good question.  What do you think of the following proposal?

* I can't immediately identify when it'd be desirable though I feel like it's reasonable to provide user control.
* Rather than expose additional hooks, an alternative is to default `compile_check` to `verifyProgramVersion`.  
  * If the caller overrides `compile_check`, it is the caller's responsibility to provide `verifyProgramVersion` as needed.
  * The proposal changes _no_ default behaviors while providing more flexibility, at the cost of requiring the caller to know which behaviors to configure.  It feels like a reasonable tradeoff to me, but I'm open to discussion.

Reply:
I agree that this solution balances all the considerations well. There may be a future use case in which a `MaybeValue` consumer wants explicitly to _not_ check program version. The flip side is that it may be unintuitive to remember to explicitly check the program version in one's custom compile checker. This problem is a small one given that we've documented that behavior.

Reply:
@jasonpaulos I'd be curious to hear your thoughts on this.

Reply:
I'll hold for a 3rd opinion.  Ready for approval once we resolve the thread.

I don't feel _too_ strongly provided all reviewers understand that the choice made here will be difficult to change later (backwards compatibility).

Reply:
With the latest update to the `compile_check` docstring, I think the current change made is acceptable (if the caller overrides `compile_check`, it is the caller's responsibility to provide verifyProgramVersion as needed).

Reply:
Gotcha - I'll consider the topic agreed on from my side.